### PR TITLE
Issue/5107 add change password

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add a ***Change Password*** option to ***Account Settings*** on the ***Me*** tab as described in https://github.com/wordpress-mobile/WordPress-Android/issues/5107.  See the animation below for illustration.

The progress dialog was included for slow network connections to block the user from exiting the ***Account Settings*** screen before the password change completes, which could leave their account and the app in a bad state.  The confirmation message was included to reassure the user that their password has been successfully changed since there is no visual verification on the ***Account Settings*** screen once the progress dialog is dismissed like the other preferences.

This pull request also includes changes to how edit text preferences with validation work.  Previously, users were allowed to enter anything and tap the ***OK*** button.  If the input was invalid, an error message was shown.  With these changes, users aren't allowed to tap the ***OK*** button until the input is valid.  This mimics the behavior while entering the address when deleting a site in the ***My Site*** > ***Settings*** > ***Delete Site*** dialog.  See the screenshots below for illustration.

### Test
1. Go to ***Me*** tab.
2. Tap ***Account Settings*** option.
3. Tap ***Change password*** item.
4. Enter zero to five characters.
5. Notice ***OK*** button is disabled.
6. Enter six or more characters.
7. Notice ***OK*** button is enabled.
8. Tap ***OK*** button.
9. Notice ***Changing password...*** progress dialog is shown.
10. Notice ***Password changed successfully*** message is shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### CODE REVIEW

**Comment by theck13** [comment]
> Error messages. There is a server-side validation for password strength, which will reject a lot of things. I was wondering why I can't change my password and it was because it's "easy to guess" or "You've recently used this password". I was only able to figure this out in Calypso, app returns
generic error. Error messages, in this case, are important, and we should probably show them.

I added showing messages for password errors in 859a504, but they will be empty until the [FluxC counterpart](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/978) is merged and the dependency hash is updated.

> Password confirmation or "show password" option. If you make a mistake with password input you might have trouble accessing your account again. If we can prevent it at the source, it would be fantastic.

I thought about the same thing while making these changes.  I wanted to add a password visibility toggle button to the input field similar to our login interface.  Unfortunately, there is no option for a second input field for confirmation or a password toggle view with `EditTextPreference`.  I tried to balance convenience and security with the current implementation by using the default behavior of `PasswordTransformationMethod`, which shows the most recently typed character momentarily for confirmation and hides previously typed characters for security.

If a user makes a mistake entering their password and they still have access to the app, they can use the app to change their password again.  If a user makes a mistake entering their password and they don't have access to the app, they can use the web to change their password via email.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AccountSettingsFragment.java [hunk 1]
```diff
@@ -275,29 +277,37 @@ public void onAccountChanged(OnAccountChanged event) {
...
-                showChangePasswordProgressDialog(false);
+            if (event.isError()) {
+                ToastUtils.showToast(getActivity(), event.error.message, ToastUtils.Duration.LONG);
+                AppLog.e(T.SETTINGS, event.error.message);
+            } else {
                 ToastUtils.showToast(getActivity(), R.string.change_password_confirmation, ToastUtils.Duration.LONG);
+                refreshAccountDetails();
             }
+        } else {
+            showChangePasswordProgressDialog(false);
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/AccountSettingsFragment.java]: Do we need to hide password change progress dialog here? If I'm understanding this correctly, `mChangePasswordProgressDialog` will always be null in this case.

Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AccountSettingsFragment.java]: Yep, when I added the `if (mChangePasswordProgressDialog != null)` check in be72326, it made the `showChangePasswordProgressDialog(false);` statement in the `else` branch unnecessary.  I remove that statement in e5de395.

**Comment by SylvesterWilmott** [comment]
LGTM @theck13 

>shows the most recently typed character momentarily for confirmation and hides previously typed characters for security.

I think this works great

**Comment by khaykov** [comment]
Looks good! I think we can merge it after resolving conflicts.

**Pull request merged**